compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -832,6 +832,7 @@ trait Checking {
             val UnApply(fn, _, _) = pat: @unchecked
             fn match
               case Select(id, _) => id
+              case Apply(Select(id, _), _) => id
               case TypeApply(Select(id, _), _) => id
           em"pattern binding uses refutable extractor `$extractor`"
```

Comment:
```suggestion
            tpd.funPart(fn) match
              case Select(id, _) => id
              case _ => ""
          if extractor == ""
            em"pattern binding uses refutable extractor"
          else
            em"pattern binding uses refutable extractor `$extractor`"
```

Reply:
oooh, was unware of `funPart`, very nice, thanks!